src/main/scala/mesosphere/marathon/storage/repository/GcActor.scala

```diff
@@ -269,25 +308,45 @@ private[storage] trait ScanBehavior[K, C, S] { this: FSM[State, Data] with Actor
       }.map(_.filter(_._2.size > maxVersions).toMap)
     }
 
+    def podsExceedingMaxVersions(usedPods: Iterable[PathId]): Future[Map[PathId, Set[OffsetDateTime]]] = {
+      Future.sequence {
+        usedPods.map { id =>
+          podRepository.versions(id).runWith(Sink.sortedSet).map(id -> _)
+        }
+      }.map(_.filter(_._2.size > maxVersions).toMap)
+    }
+
     async {
       val inUseRootFuture = rootsInUse()
       val allAppIdsFuture = appRepository.ids().runWith(Sink.set)
+      val allPodIdsFuture = podRepository.ids().runWith(Sink.set)
```

Comment:
seems like there's a pattern here. useful to refactor into a more abstract form? or maybe later if we add support for other types of persisted things (like v3 assemblies?)